packages/extension-sample/src/setup-app.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import loadable from '@loadable/component'
+import {IApplicationExtension, IRouteConfig} from '@salesforce/pwa-kit-react-sdk/ssr/universal/extensibility/types'
+import {IConfig} from './types'
+import withRedBorder from './components/with-red-border'
+
+const SamplePage = loadable(() => import('./pages/sample'))
+
+const defaultPath: string = '/sample-page'
+class Sample implements IApplicationExtension {
+    private config: IConfig;
+
+    constructor(config: IConfig) {
+        this.config = config
+    }
+    
+    getName(): string {
+        return 'sample'
+    }
+
+    extendApp(App: React.ComponentType): React.ComponentType {
+        return withRedBorder(App)
+    }
```

Comment:
would we ever want to support `tsx` in this file? if so, we should support both `setup-app.ts` and `setup-app.tsx`

For example,

```suggestion
    extendApp(App: React.ComponentType): React.ComponentType {
          return <MyProvider><App/></MyProvider>
    }
```

Reply:
This has gone through my head. But really only supported the types based on what I was emulating from your work with the server stuff. Do you think we should do that now? Or we can also leave it in to see what kind of feedback we get.. I'm ok either way